### DISCUSSION TITLE
Receiving tag information of a VM

### DISCUSSION THREAD

**Discussion by rishichordia** [vmware/govmomi#2879] [Community Help] [answered]
Hi,

I wanted to know the use of the tag field in VirtualMachine.

I've tried to retrieve the information this field contains by using Retrieve function and it has no information stored in it.

And in issue#1825, it is mentioned that tags are stored in a different model by vCenter, then what is the purpose of this tag field?

Thanks in advance!

**Comment by embano1** [comment, marked as answer]
Not 100% sure since I don't have all the background, but this field might not be used/correctly set in recent vSphere versions anymore since the tagging API is available under the [VMODL2 APIs](https://developer.vmware.com/apis/vsphere-automation/latest/cis/api/cis/tagging/tag/tag_id/get/).

`govmomi` and `govc` support this API, i.e. you can retrieve and manage tags.

VMODL1 `The set of tags associated with this managed entity. Experimental. Subject to change.`
